active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,220 @@
+    Header="Enter equation:" 
+    PlaceholderText="A + B * C" 
+    AcceptsCalculation="True" />
+```
+
+### Add formatting
+
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for IP addresss" 
+    Header="Enter equation:" 
+    PlaceholderText="000.000.000.000"
+    FormatType="IPAddress" />
+```
+
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for international telephone" 
+    Header="Enter equation:" 
+    PlaceholderText="0 (123) 456-7890"
+    FormatType="InternationalTelephone" />
+```
+
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for custom input" 
+    Header="Enter equation:" 
+    PlaceholderText="00 ft 00 in"
+    CustomFormatString="## ft ## in"
+    StripLeadingZeros="False" />
+```
+
+### Add increment and decrement stepping
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for items in order" 
+    Header="Order:" 
+    Text="2"
+    StepSize="2"
+    UpDownArrowsEnabled="True"
+    UpDownDragEnabled="True" />
+```
+
+### Add a prefix and suffix 
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for IP address" 
+    Header="Enter equation:" 
+    PlaceholderText="000.000.000.000" 
+    Prefix="IP Address:" />
+```
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for cents" 
+    Header="Enter equation:" 
+    PlaceholderText="0.00" 
+    Suffix="cents" />
+```
+
+### Add mininum and maximum values
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for items in order" 
+    Header="Order:" 
+    PlaceholderText="1" 
+    MinValue="1"
+    MaxValue="99" />
+```
+
+## Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+## API Notes
+<!-- Option 1: Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. For properties, specify the default value of the property if it
+isn't the type's default (for example an int-typed property that doesn't default to zero.) -->
+
+<!-- Option 2: Put these descriptions in the below API Details section,
+with a "///" comment above the member or type. -->
+
+## API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+```c++ 
+enum NumberBoxFormatType
+{
+    IPAddress,
+    InternationalTelephone,
+    Currency,
+};
+
+unsealed runtimeclass NumberBox : Windows.UI.Xaml.Controls.TextBox
+{
+    NumberBox();
+    
+    String Prefix;
+    String Suffix;
+    
+    Boolean AcceptsCalculation;
+    Boolean UpDownArrowsEnabled;
+    Boolean UpDownDragEnabled;
```

Comment:
How will "UpDownDrag" work?
Will it be limited to being within the control?
How much will the value change based on the amount dragged?
What happens if this control is inside a scrollable container?
Will this override current drag behavior of selected text within the control? If so, how?

Reply:
I know Xbox has the concept of a focus trap, but I don't believe this is a thing on the desktop.  Ideally with the box in focus, it would trap the drag motion.

Reply:
@mdtauk Then that would break other functionality of a standard TextBox such as selecting text and dragging it elsewhere.
I'm trying to highlight that this is something that the spec needs to cover. 
If there are precedents elsewhere then that's great because there are lots of potential edge cases here that need to be considered, understood, and documented.

Reply:
I'm assuming the `UpDownDragEnabled` actually refers to the dragging of the arrows (as you say, dragging of text doesn't make much sense). If this is the case, perhaps it should be renamed to `UpDownArrowsDragEnabled`? Dragging of the arrows would actually be pretty useful, many apps already manually implement this.

Reply:
@mrlacey Microsoft Blend uses this drag mechanic with its property TextFields.  It also changes the cursor to a four way arrow to indicate it behaves differently.

It could also be a slider control built into a TextField.  If there is no way to implement the drag mechanic, we could do what Adobe does and include a dropdown/flyout with a slider.
![image](https://user-images.githubusercontent.com/7389110/58650407-7ce5fb00-8306-11e9-8bad-5bf56e5d87fe.png)

Reply:
There is a way to do it with just mouse drag - see WinRT XAML Toolkit's `NumericUpDown` again
![266B14D2-4A29-4A3B-93F6-1D05C255A51D](https://user-images.githubusercontent.com/1753264/58669048-be1ee080-82ef-11e9-9b08-652c29efb405.GIF)

Reply:
> There is a way to do it with just mouse drag - see WinRT XAML Toolkit's `NumericUpDown` again
> ![266B14D2-4A29-4A3B-93F6-1D05C255A51D](https://user-images.githubusercontent.com/1753264/58669048-be1ee080-82ef-11e9-9b08-652c29efb405.GIF)

@xyzzer  This is how the Blend boxes work.  But I think there may be some accessibility concerns, but thats for the devs to sort out lol

Reply:
I have added the [API Note](https://github.com/microsoft/microsoft-ui-xaml-specs/commit/b7786665bb18f18d02a6133abedaa8c6d9300fbd) to align with WinRT XAML Toolkit's NumericUpDown. Thanks for surfacing this, @xyzzer, it will enable a consistent experience and help us be more effective!